libcpuid/cpuid_main.c

```diff
@@ -1748,7 +1748,7 @@ hypervisor_vendor_t cpuid_get_hypervisor(struct cpu_raw_data_t* raw, struct cpu_
 	};
 
 	if (!data) {
-		if ((r = cpu_identify(raw, data)) < 0)
+		if ((r = cpu_identify(raw, &mydata)) < 0)
 			return HYPERVISOR_UNKNOWN;
 		data = &mydata;
 	}
```

Comment:
Maybe we should just put `data = &mydata` at the beginning of the if() clause.

Reply:
```suggestion
	if (!data) {
		data = &mydata;
		if ((r = cpu_identify(raw, data)) < 0)
			return HYPERVISOR_UNKNOWN;
	}
```

Reply:
After further consideration, if I read your code like pseudo-code, it says something like "if `data` is not a valid pointer, then call `cpu_identify()` with my own valid `mydata`", so it also make sense.